cmd/samples/http/sender-with-custom-client/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/google/uuid"
+	"github.com/kelseyhightower/envconfig"
+)
+
+const (
+	count = 1000
+)
+
+type envConfig struct {
+	// Target URL where to send cloudevents
+	Target string `envconfig:"TARGET" default:"https://localhost:8080" required:"true"`
+}
+
+func main() {
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Printf("[ERROR] Failed to process env var: %s", err)
+		os.Exit(1)
+	}
+	os.Exit(_main(os.Args[1:], env))
+}
+
+// Example is a basic data struct.
+type Example struct {
+	Sequence int    `json:"id"`
+	Message  string `json:"message"`
+}
+
+func _main(args []string, env envConfig) int {
+	source, err := url.Parse("https://github.com/cloudevents/sdk-go/cmd/samples/sender")
+	if err != nil {
+		log.Printf("failed to parse source url, %v", err)
+		return 1
+	}
+
+	// Add path to client certificate and key for HTTPS
+	cert, err := tls.LoadX509KeyPair("generated.crt", "generated.key")
```

Comment:
can these paths and files be passed in? How do I generate them?

Reply:
Sorry about the name `generated`. Actually, to send events to a source which is TLS enabled, these certs are needed and this can be a valid use-case to use a custom client. Surely, we could pass as ENV variables.